src/net.h

```diff
@@ -686,13 +685,15 @@ class TransportDeserializer {
     // read and deserialize data
     virtual int Read(const char *data, unsigned int bytes) = 0;
     // decomposes a message from the context
-    virtual CNetMessage GetMessage(const CMessageHeader::MessageStartChars& message_start, std::chrono::microseconds time) = 0;
+    virtual Optional<CNetMessage> GetMessage(std::chrono::microseconds time, uint32_t& out_err) = 0;
```

Comment:
IIUC, `out_err` isn't an error number like its name suggests, but rather the size in bytes of the message. Perhaps its naming could be clearer.

Reply:
I named it this way because TransportDeserializer is such a general interface.  It's possible that some other implementation of it will use `out_err` as some sort of magic number.

Right now it is only set when the method fails.  If we're going to go down the route of renaming this to something like `msg_size`, then I would prefer to always set it.

Reply:
fixed